src/webgpu/shader/validation/shader_io/workgroup_size.spec.ts

```diff
@@ -165,3 +233,68 @@ ${kWorkgroupSizeTests[t.params.attr].src}
 @compute fn main() {}`;
     t.expectCompileResult(kWorkgroupSizeTests[t.params.attr].pass, code);
   });
+
+g.test('workgroup_size_fragment_shader')
+  .desc(`Test validation of workgroup_size on fragment shader`)
+  .fn(t => {
+    const code = `
+@workgroup_size(1)
+@fragment fn main(@builtin(position) pos: vec4<f32>) {}`;
+    t.expectCompileResult(false, code);
+  });
+
+g.test('workgroup_size_vertex_shader')
+  .desc(`Test validation of workgroup_size on fragment shader`)
+  .fn(t => {
+    const code = `
+@workgroup_size(1)
+@vertex fn main() -> @builtin(position) vec4<f32> {}`;
+    t.expectCompileResult(false, code);
+  });
+
+g.test('workgroup_size_function')
+  .desc(`Test validation of workgroup_size on user function`)
+  .fn(t => {
+    const code = `
+@workgroup_size(1)
+fn my_func() {}`;
+    t.expectCompileResult(false, code);
+  });
+
+g.test('workgroup_size_const')
+  .desc(`Test validation of workgroup_size on user function`)
```

Comment:
done.

All of the other tests just use a `compute` shader, these tests change the type of entry point for the tests, or move the `workgroup_size` to something other then an entry point. It would make the test much harder to read to have all the required templating to make it generic enough to handle the rest of these tests.

Reply:
Ack